src/kernel/src/process/module/recompiler/aarch64.rs

```diff
@@ -0,0 +1,484 @@
+use super::{LabelType, NativeCode, Recompiler, RunError};
+use crate::process::module::Segment;
+use crate::process::Process;
+use iced_x86::code_asm::{CodeAssembler, CodeLabel};
+use iced_x86::{Instruction, Register};
+use std::collections::{HashMap, VecDeque};
+
+pub struct Aarch64Emitter<'input> {
+    proc: *mut Process,
+    input: &'input [u8],
+    segments: Vec<Segment>,
+    assembler: CodeAssembler,
+    jobs: VecDeque<(u64, usize, CodeLabel)>,
+    labels: HashMap<u64, CodeLabel>,
```

Comment:
These fields should be removed as they're specific to x86_64 instructions. We won't be using the same assembler.